src/components/FormValidation/FormValidation.js

```diff
@@ -0,0 +1,44 @@
+import { useState, useCallback } from "react";
+
+export const useFormWithValidation = () => {
```

Comment:
* You should put `useFormAndValidation` in a separate file  `useFormAndValidation.js` in folder `hooks` and then import it to the component that needs validation. It's not a component of React